source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -2654,10 +2655,202 @@ AdaptiveNamespaceStart
         AddInputValueToContext(renderContext, adaptiveCardElement, *numberInputControl);
     }
 
+    void XamlBuilder::HandleInlineAcion(
+        IAdaptiveRenderContext* renderContext,
+        IAdaptiveRenderArgs* renderArgs,
+        ITextBox* textBox,
+        IAdaptiveActionElement* inlineAction,
+        IUIElement** textBoxWithInlineAction)
+    {
+        ComPtr<ITextBox> localTextBox(textBox);
+        ComPtr<IAdaptiveActionElement> localInlineAction(inlineAction);
+
+        ABI::AdaptiveNamespace::ActionType actionType;
+        THROW_IF_FAILED(localInlineAction->get_ActionType(&actionType));
+
+        // Show cards are not supported for inline actions
+        if (actionType == ActionType_ShowCard)
+        {
+            THROW_IF_FAILED(renderContext->AddWarning(
+                ABI::AdaptiveNamespace::WarningStatusCode::UnsupportedValue,
+                HStringReference(L"Show card not supported for InlineAction").Get()));
+
+            THROW_IF_FAILED(localTextBox.CopyTo(textBoxWithInlineAction));
+            return;
+        }
+
+        // Create a grid to hold the text box and the action button
+        ComPtr<IGridStatics> gridStatics;
+        THROW_IF_FAILED(GetActivationFactory(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_Grid).Get(), &gridStatics));
+
+        ComPtr<IGrid> xamlGrid = XamlHelpers::CreateXamlClass<IGrid>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_Grid));
+        ComPtr<IVector<ColumnDefinition*>> columnDefinitions;
+        THROW_IF_FAILED(xamlGrid->get_ColumnDefinitions(&columnDefinitions));
+        ComPtr<IPanel> gridAsPanel;
+        THROW_IF_FAILED(xamlGrid.As(&gridAsPanel));
+
+        // Craete the first column and add the text box to it
+        ComPtr<IColumnDefinition> textBoxColumnDefinition = XamlHelpers::CreateXamlClass<IColumnDefinition>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_ColumnDefinition));
+        THROW_IF_FAILED(textBoxColumnDefinition->put_Width({ 1, GridUnitType::GridUnitType_Star }));
+        THROW_IF_FAILED(columnDefinitions->Append(textBoxColumnDefinition.Get()));
+
+        ComPtr<IUIElement> textBoxAsUIElement;
+        THROW_IF_FAILED(localTextBox.As(&textBoxAsUIElement));
```

Comment:
we don't use this until quite a bit further down... shouldn't we be declaring closer to use?